tools/ab_test

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+# -*- shell-script -*-
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -eu -o pipefail
+shopt -s lastpipe
+
+
+if [ $# -lt 3 ]; then
+    cat <<EOF
+$0 <a revision> <b revision> <A/B test glob> [pytest args]
+
+    Example: $0 f6c28a6e2c50eaf021130c3f4e59cbd8daf0ea12 bb2fa406e12e7c49f48fce770b57fe6ec856d263 integration_tests/performance/test_snapshot_ab.py
+
+    Runs the specified A/B test with the given revisions. This is useful for running ad-hoc A/B-test to investigate
+    performance impact of a specific commit range by running this script as a buildkite step.
+EOF
+    exit 1
+fi
+
+function get_binaries {
+  revision=$1
+
+  if [ -d build/$revision ]; then
+    return
+  fi
+
+  temp=$(mktemp -d)
+  git clone "https://github.com/firecracker-microvm/firecracker" $temp
+  cd $temp
+  ./tools/devtool build --release
```

Comment:
Shouldn't we run `git checkout $revision` before this?

Reply:
... oh wow, I really hope this is something I broken when untangling my code into commits, otherwise I've run some really useless tests :joy: